chore: Add Azure OpenAI embedding provider

shubhiroy opened the pull request:
Added support for azure open ai embedding model.

drazvan commented on nemoguardrails/embeddings/providers/azureopenai.py:
```diff
@@ -0,0 +1,105 @@
+        """Retrieve the embedding dimension for the specified model."""
+        if self.embedding_model in self.MODEL_DIMENSIONS:
+            return self.MODEL_DIMENSIONS[self.embedding_model]
+        else:
```
We should fall back to computing the embedding for a test string, and take the size from there. Otherwise, if a new model is published, it will not be usable until a new version of nemoguardrails is published.

drazvan replied:
@shubhiroy : let me know if you can address these changes in the next few days, so we can include this in the 0.10.0 release. Thanks!

shubhiroy replied:
Hi @drazvan,
I have done the required changes.

1. Modified code to compute embedding using test string
2. Added details in configuration-guide.md
3. Added test

Sorry for late reply. I was not well from last few weeks.

drazvan reviewed (dismissed):
Looks good :+1:. You should update the `configuration-guide.md` file to mention this as well. Also, please add a minimal test for this. Thanks!

shubhiroy commented:
Hi @drazvan,
I have the required changes. 

1. Modified code to compute embedding using test string
2. Added details in configuration-guide.md
3. Added test

Sorry for late reply. I was not well from last few weeks.

Pouyanpi commented:
@shubhiroy it seems that your PR was pending review for a while, thanks for being patient 👍🏻 

Is it possible for you to rebase it to develop branch? I'm not sure if you have noticed but the repo has changed quite significantly since the PR was opened. Also dev workflow has changed a bit (see [contributing guide](https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/CONTRIBUTING.md)). Let's start by resolving the conflict first.

trebedea reviewed (approved):
Looks good.
